### PR TITLE
refactor vpa to use helm-addon module

### PR DESCRIPTION
**Pull request by askulkarni2**
### What does this PR do?

Refactor VPA addon to use the helm-addon module.

### Motivation

Standardize addons.

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [N/A] Yes, I have added a new example under [examples](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/examples) to support my PR
- [N/A] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/ssp-eks-add-ons) repo (if applicable)
- [N/A] Yes, I have updated the [docs](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/docs) for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [X] E2E Test successfully complete before merge?

### CODE REVIEW

File: modules/kubernetes-addons/vpa/locals.tf [hunk 1]
```diff
@@ -1,71 +1,38 @@
...
+    kubernetes_namespace              = local.helm_config["namespace"]
+    kubernetes_service_account        = local.service_account_name
+    create_kubernetes_namespace       = true
+    create_kubernetes_service_account = true
+    iam_role_path                     = "/"
+    tags                              = var.tags
+    eks_cluster_id                    = var.eks_cluster_id
+    irsa_iam_policies                 = var.irsa_policies
+    irsa_iam_permissions_boundary     = var.irsa_permissions_boundary
+  }
```
Comment (vara-bonthu) [inline modules/kubernetes-addons/vpa/locals.tf:35]: I don't think we need IRSA for VPA. Are you creating any policies here?

File: modules/kubernetes-addons/vpa/variables.tf [hunk 2]
```diff
@@ -1,29 +1,34 @@
...
+  type        = list(string)
+  default     = []
+  description = "IAM Policy ARN list for any IRSA policies"
+}
+
+variable "irsa_permissions_boundary" {
+  type        = string
+  default     = ""
+  description = "IAM Policy ARN for IRSA IAM role permissions boundary"
+}
```
Comment (vara-bonthu) [inline modules/kubernetes-addons/vpa/variables.tf:34]: We may not need this. See the comment above

**Review by vara-bonthu — CHANGES_REQUESTED** [review]
Nice! Few comments requires change 👍🏼

**Review by kcoleman731 — COMMENTED** [review]
Looks good to me. Just verify @vara-bonthu comments.

File: modules/kubernetes-addons/vpa/locals.tf (hunk 1 above)
Comment (askulkarni2) [reply modules/kubernetes-addons/vpa/locals.tf]: addressed.

File: modules/kubernetes-addons/vpa/main.tf [hunk 3]
```diff
@@ -1,77 +1,6 @@
...
-    content {
-      name  = each_item.value.name
-      value = each_item.value.value
-    }
-  }
+module "helm_addon" {
+  source            = "../helm-addon"
+  manage_via_gitops = var.manage_via_gitops
+  helm_config       = local.helm_config
+  irsa_config       = local.irsa_config
```
[the remark this answers is not in the document]
Comment (askulkarni2) [reply modules/kubernetes-addons/vpa/main.tf:5]: we still need the namespace and service account. Removed IAM stuff.

**Review by kcoleman731 — COMMENTED** [review]
LGTM @vara-bonthu please review in the AM.

**Review by vara-bonthu — CHANGES_REQUESTED** [review]
LGTM! One change required

**Review by vara-bonthu — APPROVED** [review]

**Review by kcoleman731 — APPROVED** [review]

**Review by Zvikan — APPROVED** [review]

**Pull request merged**